### PR TITLE
Separate my-ocular, add Aviate functionality

### LINKED ISSUE #4706: Addon Request: Aviate Integration
### What is Aviate
Aviate is an Ocular alternative created by [MystPi](https://github.com/MystPi)\([NFlex23](https://scratch.mit.edu/users/NFlex23/)\). They recently [created an extension for Chromium based browsers](https://github.com/MystPi/aviate-integration) which adds Aviate statues under your name in the forums(just like the ocular integration addon).

### PR DESCRIPTION
**Pull request by mybearworld**
Resolves #4706

### Changes

Separate `my-ocular` into two addons.

- `my-ocular`, which is now responsible for just reactions and the "view on ocular" links
- `user-status`, which is now responsible for statuses and also works with Aviate

Demo

### Reason for changes

To integrate well with Aviate.

### Tests

Tested in Edge.

### CODE REVIEW

**Comment by mybearworld** [comment]
@MystPi Would it be possible to set up CORS on the API such that other sites can access it easily?

**Comment by MystPi** [comment]
> @MystPi Would it be possible to set up CORS on the API such that other sites can access it easily?

Somehow I missed that in the rewrite, but I think it should work now.

**Comment by mybearworld** [comment]
> > @MystPi Would it be possible to set up CORS on the API such that other sites can access it easily?
> 
> Somehow I missed that in the rewrite, but I think it should work now.

It does!

**Comment by GrahamSH-LLK** [comment]
There is no chance we're getting rid of Ocular statuses. First of all, Ocular is moderated by SA community members so we can actually trust what the extension is showing. Second of all, it's horribly confusing for users to have either-or, since the two systems are incompatible. If they want Aviate, they can install STE.

**Comment by mybearworld** [comment]
> Second of all, it's horribly confusing for users to have either-or, since the two systems are incompatible.

It's not either/or. You can have both.

**Comment by mybearworld** [comment]
> Ocular is moderated by SA community members so we can actually trust what the extension is showing.

What are you trying to say? The notion that everyone who isn't a SA community member isn't trustworthy js not healthy.

**Comment by Samq64** [comment]
I think offering both options should be fine as long as both can't be active at the same time.

**Comment by mybearworld** [comment]
> I think offering both options should be fine as long as both can't be active at the same time because having 2 status is confusing.

But why artificially limit it?

**Comment by Samq64** [comment]
I guess one option would be to check if the user has an Aviate status and fallback to Ocular if both are enabled.

**Comment by mybearworld** [comment]
> Or just keeping both and distinguishing them more than just the coloured dot.

That would be a good option, but how else would you differentiate between them?

**Comment by Samq64** [comment]
> > Or just keeping both and distinguishing them more than just the coloured dot.
> 
> That would be a good option, but how else would you differentiate between them?

I can't think of anything other than putting Ocular or Aviate before the status and adding an extra new line between them.

**Comment by mybearworld** [comment]
I originally had an extra new line there, but it just looked really bad on profiles.

File: addons-l10n/en/user-status.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+  "user-status/status-hover": "This is a customized status from ocular, displayed by Scratch Addons. You can set your own at https://ocular.jeffalo.net/dashboard."
```
Comment (RedGuy12) [inline addons-l10n/en/user-status.json:2]: aviate should have its own msg too

Comment (mybearworld) [reply addons-l10n/en/user-status.json:2]: True

**Comment by WorldLanguages** [comment]
There's some related discussion about Aviate on #4739

> What are you trying to say? The notion that everyone who isn't a SA community member isn't trustworthy is not a good one.

I think Graham's point is that there already exists a "trustworthy" alternative that is very similar to Aviate

**Comment by mybearworld** [comment]
> I think Graham's point is that there already exists a "trustworthy" alternative that is very similar to Aviate

Well, this is a new alternative that's semi-established with a component system. Thinking in "trustworthiness" seems wrong.

**Comment by mybearworld** [comment]
> There's some related discussion about Aviate on #4739

The point there seems to be that nobody uses it - that was almost a year ago, and a lot of the ATs seem to be using it now.

**Comment by WorldLanguages** [comment]
> Thinking in "trustworthiness" seems wrong.

Well, it's something to consider as statuses are user-generated content without moderation by the Scratch Team, and it's not so clear how one would report an inappropriate status. Also, the service may be able to connect the dots and figure out which forum posts and profiles a scratcher is visiting.

**Comment by WorldLanguages** [comment]
> The point there seems to be that nobody uses it - that was almost a year ago, and a lot of the ATs seem to be using it now.

Yeah, looks like Aviate founds its way into the forumers, so we can probably figure out a way to add it to the extension.

**Comment by DNin01** [comment]
> Well, it's something to consider as statuses are user-generated content without moderation by the Scratch Team, and it's not so clear how one would report an inappropriate status.

Yeah, I think that could become a problem.

**Comment by mybearworld** [comment]
> Also, the service may be able to connect the dots and figure out which forum posts and profiles a scratcher is visiting.

ScratchDB also has that risk, but I do think we can assume that that's not the case.

> *somehow... I never understand forumers 😂 the weirdest addons are also forum addons

Very true :P

> Well, it's something to consider as statuses are user-generated content without moderation by the Scratch Team, and it's not so clear how one would report an inappropriate status.

I suppose to NFLex23/MystPi - though this is assuming most users will be operating in good faith. Same problem with Ocular, where the "trustworthiness" comes into play again.

**Comment by MystPi** [comment]
I plan to add a "report" page very soon. It will allow you to enter a user's username with an additional description of why you are reporting them. As of now users can contact me on my Scratch profile, but I am aware this is suboptimal.

There will be a link to the new page on the dashboard and the FAQs page. Additionally, if user pages are also added, there will be a section for reporting the status there as well.

**Comment by mybearworld** [comment]
> I plan to add a "report" page very soon. It will allow you to enter a user's username with an additional description of why you are reporting them. As of now users can contact me on my Scratch profile, but I am aware this is suboptimal.

That sounds a little problematic as it would essentially be private messaging - maybe you should make it so that you get a dropdown list of users to choose from, and another one where you can also pick yourself from that list so you can reach out to them in case you don't know why they are reporting?

**Comment by MystPi** [comment]
> > I plan to add a "report" page very soon. It will allow you to enter a user's username with an additional description of why you are reporting them. As of now users can contact me on my Scratch profile, but I am aware this is suboptimal.
> 
> That sounds a little problematic as it would essentially be private messaging - maybe you should make it so that you get a dropdown list of users to choose from, and another one where you can also pick yourself from that list so you can reach out to them in case you don't know why they are reporting?

You're right. A dropdown for yourself isn't necessary because Aviate knows what user is logged in. However, a dropdown that contains every user would be very laggy as there are 750+ users. Maybe it should just validate that the user you are reporting exists.

**Comment by mybearworld** [comment]
> A dropdown for yourself isn't necessary because Aviate knows what user is logged in.

True, forgot about that :P

> However, a dropdown that contains every user would be very laggy as there are 750+ users.

Might be doable using a `datalist`...

File: addons/user-status/addon.json [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+      "name": "Show Ocular statuses",
+      "id": "ocular",
+      "type": "boolean",
+      "default": true
+    },
+    {
+      "name": "Show Aviate statuses",
+      "id": "aviate",
+      "type": "boolean",
+      "default": true
```
Comment (WorldLanguages) [inline addons/user-status/addon.json:39]: `"default": true` on Aviate probably won't make it

Comment (mybearworld) [reply addons/user-status/addon.json:39]: Fair

```suggestion
      "default": false
```

**Comment by Secret-chest** [comment]
Both ocular and Aviate should be available, we could differentiate by showing the logos. We should have the following options:
* Prefer ocular
* Prefer Aviate
* Only ocular
* Only Aviate
* ocular first
* Aviate first

**Comment by mybearworld** [comment]
Slight problem with ToasterUnited's statuses - not sure why that's happening:

image

**Comment by mybearworld** [comment]
> Don't use `innerHTML`, BTW

I changed the way it works in commit https://github.com/ScratchAddons/ScratchAddons/pull/6473/commits/b75efcdadfa7acd5e658645b0ec8ef21b46420c9, it doesn't use innerHTML anymore

**Comment by mybearworld** [comment]
Fixed.
image

**Comment by mybearworld** [comment]
> Both ocular and Aviate should be available, we could differentiate by showing the logos. We should have the following options:
> 
> * Prefer ocular
> * Prefer Aviate
> * Only ocular
> * Only Aviate
> * ocular first
> * Aviate first

"Only ocular", "Only Aviate" and "Both" are the current options. What is the point of using the "Prefer" ones? Generally, when multiple sources of information are available, I'd rather have both then choose one. Regarding the "First" settings, is it really worth creating settings for that?

**Comment by Secret-chest** [comment]
I feel the logos are too large. They should be an icon before the status, not beside it.

**Comment by MystPi** [comment]
> I feel the logos are too large. They should be an icon before the status, not beside it.

Do you mean the logos on top of the statuses? I think it would look worse that way.

**Comment by jeffalo** [comment]
I know I'm late to this discussion but I just wanted to share my thoughts. There shouldn't be 2 "status systems" in SA. Aviate is awesome - but I think it's a bit unnecessary, and I honestly don't think it adds enough value to justify fracturing a relatively small user base (there's only ~5000 users between ocular and aviate) - adding it to SA would only be confusing... The only differentiatiating feature is the components, which while cool, could have just been PR'd into ocular, which would avoid having to figure out UX for all of this. I'd be down to handover write perms to ocular's status system to NFlex if they're also down for that...

On another note, not enough people are talking about. "Status systems" are a gray area on Scratch. They fall just on the line of "unmoderated chat". ocular's gotten by because statuses are just a small part of the site, but I'm not sure how the Scratch Team would react to people discussing Aviate statuses (outside of the forum thread) now.

**Comment by MystPi** [comment]
This is a lot to think about!

I will say, the first time I heard of this idea I was quite averse to it, but now—with a more open mind—I am beginning to understand your reasoning.

I do have a few questions/concerns as of right now...
- Would the status system be an npm package or be embedded in ocular itelf?
- Where would the component reference be, and how would it be updated? Currently Aviate loops through all of the components and uses their descriptions—I think ocular would be able to do the same.
- Similarly, would there be a tutorial page like [this one](https://aviate.scratchers.tech/tutorial)?
- Migrating all of the existing statuses would be a pain.

**Comment by WorldLanguages** [comment]
> On another note, not enough people are talking about. "Status systems" are a gray area on Scratch. They fall just on the line of "unmoderated chat". ocular's gotten by because statuses are just a small part of the site, but I'm not sure how the Scratch Team would react to people discussing Aviate statuses (outside of the forum thread) now.

You have a point, but it doesn't seem like whether this is merged or not would affect what you're saying. The harm is done already.

Also, please try discussing non addon details somewhere else @MystPi

**Comment by BroJac5246** [comment]
> > Both ocular and Aviate should be available, we could differentiate by showing the logos. We should have the following options:
> > 
> > * Prefer ocular
> > * Prefer Aviate
> > * Only ocular
> > * Only Aviate
> > * ocular first
> > * Aviate first
> 
> "Only ocular", "Only Aviate" and "Both" are the current options. What is the point of using the "Prefer" ones? Generally, when multiple sources of information are available, I'd rather have both then choose one. Regarding the "First" settings, is it really worth creating settings for that?

I believe "prefer" is "only use one, but if both exist then use this one." That is what I was imagining for this addon.

**Comment by mybearworld** [comment]
> I believe "prefer" is "only use one, but if both exist then use this one." That is what I was imagining for this addon.

I know, that's what the next sentence is referring to:

> Generally, when multiple sources of information are available, I'd rather have both then choose one.

**Comment by BroJac5246** [comment]
> Both ocular and Aviate should be available, we could differentiate by showing the logos. We should have the following options:
> * Prefer ocular
> * Prefer Aviate
> * Only ocular
> * Only Aviate
> * ocular first
> * Aviate first

@WorldLanguages what is your reasoning for the laughing face emoji? This solution makes sense and covers all combinations of Ocular and Aviate that a user could want.

**Comment by mybearworld** [comment]
> This solution makes sense and covers all combinations of Ocular and Aviate that a user could want.

I'm not WL, but it just seems like a lot of options that aren't too important.

**Comment by mxmou** [comment]
> > On another note, not enough people are talking about. "Status systems" are a gray area on Scratch. They fall just on the line of "unmoderated chat". ocular's gotten by because statuses are just a small part of the site, but I'm not sure how the Scratch Team would react to people discussing Aviate statuses (outside of the forum thread) now.
> 
> You have a point, but it doesn't seem like whether this is merged or not would affect what you're saying. The harm is done already.

A lot of new users would probably join Aviate if we release an addon that displays Aviate statuses.

**Comment by BroJac5246** [comment]
> On another note, not enough people are talking about. \"Status systems\" are a gray area on Scratch. They fall just on the line of \"unmoderated chat\".

Is Aviate unmoderated? It has a reporting system in place.

**Comment by rgantzos** [comment]
> 

“Unmoderated” means not moderated by the Scratch Team. If the Scratch Team decides that a specific platform (such as Ocular) is harmless, then they’ll allow it. [More info](https://scratch.mit.edu/discuss/topic/357609/?page=1#post-3622655)

**Comment by Secret-chest** [comment]
@rgantzos that only applies to things on Scratch itself. They allow any site as long as it's appropiate and moderated by anyone.

**Comment by rgantzos** [comment]
> @rgantzos that only applies to things on Scratch itself. They allow any site as long as it's appropiate and moderated by anyone.

I believe I read somewhere that they generally allow things until they cause more harm than good.

**Comment by mxmou** [comment]
> > On another note, not enough people are talking about. "Status systems" are a gray area on Scratch. They fall just on the line of "unmoderated chat".
> 
> Is Aviate unmoderated? It has a reporting system in place.

> @rgantzos that only applies to things on Scratch itself. They allow any site as long as it's appropiate and moderated by anyone.

I remember a ST member saying that sites moderated by Scratchers don't count as moderated.

**Comment by WorldLanguages** [comment]
In the community guidelines, "unmoderated" doesn't mean "not moderated" (as stupid as this sounds). Of course, linking to a moderated Discord server or YouTube livestream chat still counts as "unmoderated chat" to the Scratch Team.

**Comment by DNin01** [comment]
> > > On another note, not enough people are talking about. "Status systems" are a gray area on Scratch. They fall just on the line of "unmoderated chat".
> > 
> > 
> > Is Aviate unmoderated? It has a reporting system in place.
> 
> > @rgantzos that only applies to things on Scratch itself. They allow any site as long as it's appropiate and moderated by anyone.
> 
> I remember a ST member saying that sites moderated by Scratchers don't count as moderated.

Here's one of those comments: https://scratch.mit.edu/discuss/post/6352480/

**Comment by BroJac5246** [comment]
Two questions:

1. Is the Scratch Team likely to take issue with Aviate? It's very similar to Ocular, which we've concluded they're okay with, and it's certainly not a secret. It has a topic on the ATs.

2. Would including Aviate realistically endanger Scratch Addons? It appears to be safe, and it's basically a better version of Ocular. And why would the ST's stance on it change?

**Comment by Jazza-231** [comment]
I'm aware I have no influence in the scratch addons development, but I would like to add my views on this.
I believe that Aviate should not be added to scratch addons for various reasons. Why should scratch have even more complex statuses, when that is not a required part of scratch at all, and I barely even consider ocular relevant enough to add? Scratch at its core is about teaching children the basics of code, and it has its own problems which SA aims to fix and to overall improve the experience of using Scratch, without changing the functionality too much. Adding simple statuses is already somewhat controversial, in my eyes the "about me" and "what I'm working on" boxes should already be enough, having statuses just isn't what scratch is about. I'm sure the fact that jeffalo is a known name in the SA dev community already contributed to the fact that ocular was added. When people say that aviate is a better version of ocular, what they mean is that it is capable of more and is more complex. Do we really need to add that level of customizability and complexity to something that really isn't needed? Why is a short status and colour, something that most scratchers don't have in the first place, not enough for us? And, I read somewhere in this thread of comments that STE has aviate functionality. STE's mission is a different mission to SA, and so I think it's completely reasonable to say that if people want Aviate, they should use STE, instead of adding something that I personally, and I assume others, consider "overkill" or "unnecessary". There are also a lot of concerns regarding the ability to abuse this system. If there are too many cases of statuses that the Scratch Team consider inappropriate for the website, will they change their mind on allowing these? Are ocular and Aviate even moderated at all? I genuinely believe that adding more ways to possibly anger the scratch team to SA is a bad idea, we already do that plenty enough simply by existing. In the past, the scratch team has been very clear that unless they moderate the chat, they don't consider it to be allowable on their website. Moving back to the actual addon, the number of settings proposed to choose how to show both ocular and Aviate is much too high for such little changes visually. I agree with the many people who have said that adding 2 status systems to SA is unnecessary. Hopefully, this comment helps to inform people further on how, as someone who does use ocular statuses but isn't heavily involved with the SA dev community, feels on this topic.

**Comment by Hans5958** [comment]
> I believe that Aviate should not be added to scratch addons for various reasons. Why should scratch have even more complex statuses, when that is not a required part of scratch at all...

Your argument falls down when you try to frame it in a bigger picture. Do people really need...

- letting you see messages on an extension popup, or
- having a progress bar, or
- having these developer tools for their editor, or
- having high-definitions uploads, or
- theming feature (dark mode et al), and so on

You should know that this can apply to most, if not all addons in SA. Heck, you should have realised that SA is not an essential tool to experience whatever Scratch offers. I could just, leave it there, and still be able to use Scratch. It's an extension after all.

If you want to argue that there are some addon that are more useful than the others, and you should realise that your argument is subjective. At that point what matters more is if anyone is around to maintain these addons, not the popularity of them.

> And, I read somewhere in this thread of comments that STE has aviate functionality. STE's mission is a different mission to SA...

As of STE and SA, I still fail to understand any difference of purpose. I never used STE and I rarely visit Scratch to experience SA, but I thought both complements the experience of Scratch. I think this is what outsiders also think when facing these two extensions.

> There are also a lot of concerns regarding the ability to abuse this system. If there are too many cases of statuses that the Scratch Team consider inappropriate for the website, will they change their mind on allowing these? ...

Reminder that having *ocular* is a huge problem, and adding Aviate doesn't change anything about it due to the overlapping scope. This argument would appropriate if we want to remove *ocular* in the first place. I don't want this argument to be reason to deny Aviate, when we already have *ocular*.

> ...the number of settings proposed to choose how to show both ocular and Aviate is much too high for such little changes visually. ...

As of the problem of is it it to have two systems, I still fail to understand why this is bothering to some people. I saw the source code and it is quite simple to maintain and I only see two new settings which is a simple addition for me, (Note that yes, this is subjective, but I just want to see why)

**Comment by Samq64** [comment]
> What's an STE?

ScratchTools. The E stands for extension so it isn't mistaken for Scratch team.

**Comment by Secret-chest** [comment]
> What's an STE?
> 
> ScratchTools. The E stands for extension so it isn't mistaken for Scratch team.

Ah, that SA rip-off.

**Comment by callumjt** [comment]
> > ScratchTools. The E stands for extension so it isn't mistaken for Scratch team.
> > Ah, that SA rip-off.
> > ._.
> 
> No offence, but you should submit your addons to SA.

if i make something that would be accepted, i would make it for both. i dont see why you should neglect putting your work on another extension

**Comment by mybearworld** [comment]
> I think you possibly understood the point backwards? Scratch Addons is already "banned" from being mentioned in the Scratch website.

They could make it straight-up disallowed to use, though. The danger is there, but I don't think an Aviate addon would trigger that.

> And, I read somewhere in this thread of comments that STE has aviate functionality. STE's mission is a different mission to SA

While STE's mission does differ from SA's, there is a significant overlap (in particular, any SA addon could be an STE addon), so just because an addon is an STE addon, doesn't mean it can't be an SA one.

**Comment by BroJac5246** [comment]
> > I think you possibly understood the point backwards? Scratch Addons is already "banned" from being mentioned in the Scratch website.
> 
> They could make it straight-up disallowed to use, though. The danger is there, but I don't think an Aviate addon would trigger that.

Note: this is what I was getting at, and I agree with mybearworld.

**Comment by BroJac5246** [comment]
Here are some of the best points that have been brought up so far (both for and against this):

> > Ocular is moderated by SA community members so we can actually trust what the extension is showing.
> 
> What are you trying to say? The notion that everyone who isn't a SA community member isn't trustworthy is not a good one.

> I know I'm late to this discussion but I just wanted to share my thoughts. There shouldn't be 2 "status systems" in SA. Aviate is awesome - but I think it's a bit unnecessary, and I honestly don't think it adds enough value to justify fracturing a relatively small user base (there's only ~5000 users between ocular and aviate) - adding it to SA would only be confusing... The only differentiatiating feature is the components, which while cool, could have just been PR'd into ocular, which would avoid having to figure out UX for all of this. I'd be down to handover write perms to ocular's status system to NFlex if they're also down for that...

> In the community guidelines, "unmoderated" doesn't mean "not moderated" (as stupid as this sounds). Of course, linking to a moderated Discord server or YouTube livestream chat still counts as "unmoderated chat" to the Scratch Team.

> **750+ users**

> > There are also a lot of concerns regarding the ability to abuse this system. If there are too many cases of statuses that the Scratch Team consider inappropriate for the website, will they change their mind on allowing these? ...
> 
> Reminder that having _ocular_ is a huge problem, and adding Aviate doesn't change anything about it due to the overlapping scope. This argument would appropriate if we want to remove _ocular_ in the first place. I don't want this argument to be reason to deny Aviate, when we already have _ocular_.
> 
> > ...the number of settings proposed to choose how to show both ocular and Aviate is much too high for such little changes visually. ...
> 
> As of the problem of is it it to have two systems, I still fail to understand why this is bothering to some people. I saw the source code and it is quite simple to maintain and I only see two new settings which is a simple addition for me, (Note that yes, this is subjective, but I just want to see why)

From #4739:
> I agree with WL, nobody uses it now because there's no easy way to see someone's status, and I don't think we want to turn in to a big advertisement hub for new Scratch sites. Plus, the only advantage Aviate has over Ocular is the tags system which I feel is overkill - how much dynamics do you need? (If you really want it, PR it into Ocluar?)

My thoughts:
It looks like the best solution would be to merge Aviate into Ocular, including the Components feature, since Aviate has a tiny user base. This would solve most of the issues brought up here and would unify the two systems. @WorldLanguages you mentioned that this is being considered, I think I missed that. Can you point out where it was indicated that a merge is being considered? Thanks :D

**Comment by WorldLanguages** [comment]
> Also, please be patient as apparently Ocular and Aviate may somehow merge sometime soon, making it unnecessary to support both services from our end (and also meaning adding Aviate support wouldn't involve privacy concerns)

Just be patient folks. Don't make me lock this thread.

**Comment by mybearworld** [comment]
> > Also, please be patient as apparently Ocular and Aviate may somehow merge sometime soon, making it unnecessary to support both services from our end (and also meaning adding Aviate support wouldn't involve privacy concerns)
> 
> Just be patient folks. Don't make me lock this thread.

Why can't we discuss this? Merging Aviate into ocular was somewhat of an idea since the beginning of Aviate and never went anywhere. If ocular does get Aviate's component system, this addition can just be removed again.

**Comment by BroJac5246** [comment]
> > Also, please be patient as apparently Ocular and Aviate may somehow merge sometime soon, making it unnecessary to support both services from our end (and also meaning adding Aviate support wouldn't involve privacy concerns)
> 
> Just be patient folks. Don't make me lock this thread.

Merging them is a solution. It seems that most people would be okay with it. Why would it result in this thread needing to be locked? I'm simply curious as to where the discussion of a merger is occurring. Either I missed it or it's occurring elsewhere.

**Comment by Secret-chest** [comment]
> if i make something that would be accepted, i would make it for both. i dont see why you should neglect putting your work on another extension

thanks for blocking me from your org

**Comment by WorldLanguages** [comment]
> Merging them is a solution. It seems that most people would be okay with it. Why would it result in this thread needing to be locked? I'm simply curious as to where the discussion of a merger is occurring. Either I missed it or it's occurring elsewhere.

I'll have to lock this for a week if yall continue with off-topic discussion.
I think discussion is happening privately between jeffalo and Aviate admins. I'd wait to see what they come up with. Whatever they decide will condition how Scratch Addons responds to it.

**Comment by mybearworld** [comment]
> > Merging them is a solution. It seems that most people would be okay with it. Why would it result in this thread needing to be locked? I'm simply curious as to where the discussion of a merger is occurring. Either I missed it or it's occurring elsewhere.
> 
> I'll have to lock this for a week if yall continue with off-topic discussion. I think discussion is happening privately between jeffalo and Aviate admins. I'd wait to see what they come up with. Whatever they decide will condition how Scratch Addons responds to it.

What exactly is "the off-topic discussion" here?

@MystPi, is this correct?

**Comment by MystPi** [comment]
> > I'll have to lock this for a week if yall continue with off-topic discussion.
> 
> What exactly is "the off-topic discussion" here?
> 
> > I think discussion is happening privately between jeffalo and Aviate admins. I'd wait to see what they come up with. Whatever they decide will condition how Scratch Addons responds to it.
> 
> @MystPi, is this correct?

There hasn't been any discussion as of right now. My ideal scenario would be to keep Aviate separate from ocular and have this addon merged into SA. If you all decide against merging this—which I'd be okay with—I'll most likely still keep Aviate going on its own unless some discussion with jeffalo clears things up.

**Comment by BroJac5246** [comment]
> > > I'll have to lock this for a week if yall continue with off-topic discussion.
> > 
> > 
> > What exactly is "the off-topic discussion" here?
> > > I think discussion is happening privately between jeffalo and Aviate admins. I'd wait to see what they come up with. Whatever they decide will condition how Scratch Addons responds to it.
> > 
> > 
> > @MystPi, is this correct?
> 
> There hasn't been any discussion as of right now. My ideal scenario would be to keep Aviate separate from ocular and have this addon merged into SA. If you all decide against merging this—which I'd be okay with—I'll most likely still keep Aviate going on its own unless some discussion with jeffalo clears things up.
> 
> I understand that this is a very controversial addon and am willing to submit to whatever you guys decide.

What is your reasoning for keeping it separate? On the surface, it appears to make more sense to merge the two, but what benefits do you see coming from keeping them separate?

@jeffalo what are your thoughts on merging Ocular and Aviate, based on both the points made in this issue and outside factors?

**Comment by WorldLanguages** [comment]
> What exactly is "the off-topic discussion" here?

These are off-topic:

- The mission of other software projects, websites or extensions.
- Discussion between Ocular and Aviate admins were already moved outside this thread, as I asked. Scratch Addons is affected, but unrelated to those discussions. If they want to talk in public, they can open an issue in this repository, but this PR is not the right place.
- Asking Ocular and Aviate admins if they had reached to a conclusion yet and ask for justification. Give them a few days.

**Comment by mybearworld** [comment]
> > What exactly is "the off-topic discussion" here?
> 
> These are off-topic:
> 
> * The mission or past intentions of other software projects, websites or extensions.

👍

> * Discussion between Ocular and Aviate admins were already moved outside this thread, as I asked. Scratch Addons is affected, but unrelated to those discussions. If they want to talk in public, they can open an issue in this repository, but this PR is not the right place.

This seems very important to whether this addon ultimately gets merged, doesn't it?

> * Asking Ocular and Aviate admins if they had reached to a conclusion yet and asking for justifications. Give them a few days.

It's not been asked if there's any conclusion, just whether there's any actual interest in the thing being concluded, which seems to not be the case. This is also pretty valuable information in whether this PR can/will get merged.

**Comment by Hans5958** [comment]
> This seems very important to whether this addon ultimately gets merged, doesn't it?

No. Best we can do is to deal what we have **now**, not in the future.

**Comment by BroJac5246** [comment]
> > This seems very important to whether this addon ultimately gets merged, doesn't it?
> 
> No. Best we can do is to deal what we have **now**, not in the future.

But this issue is https://github.com/ScratchAddons/ScratchAddons/labels/status%3A%20pending%20upstream. These decisions about merging Aviate and Ocular directly affect this PR, which is unable to move forward right _now_. At some point in the future, whenever the label is removed, they will no longer be future decisions, they will be current decisions. And then we will be "[dealing with] what we have **now**," since "now" will be what is currently "the future."

**Comment by Hans5958** [comment]
I disagree, and I may remove the label.

It seems that the label is given from the discussion of the Aviate-ocular merge, which I also think it is not related to the scope of this PR. With this I may take action by removing the label.

While it is true that the mentioned merge directly affects the implementation of the addons, this doesn't mean we have to wait. The current implementation is appropriate with the current situation, and if anything happens with both services, adjustments can be made later, either in this PR, or in later PRs.[^1]

Not to mention that the mentioned merge also would affect the currently released `my-ocular` implementation, so just like before, I don't want to this as a reason to deny this PR.

Creating an issue and discussing it there would be more appropriate instead of discussing it in this PR, since the scope of the mentioned merge is more than just this PR.

[^1]: Note that I won't expect anyone to do this adaptation, just like I don't expect anyone to create this PR. This is volunteer work, and there are no contractual obligations.

**Pull request closed without merging**